snuba/utils/streams/kafka.py

```diff
@@ -257,61 +283,41 @@ def __seek(self, offsets: Mapping[TopicPartition, int]) -> None:
                     ConfluentTopicPartition(stream.topic, stream.partition, offset)
                 )
 
-        self.__offsets.update(offsets)
+        self.__working_offsets.update(offsets)
 
     def seek(self, offsets: Mapping[TopicPartition, int]) -> None:
         if self.__state in {KafkaConsumerState.CLOSED, KafkaConsumerState.ERROR}:
             raise InvalidState(self.__state)
 
-        if offsets.keys() - self.__offsets.keys():
+        if offsets.keys() - self.__working_offsets.keys():
             raise ConsumerError("cannot seek on unassigned streams")
 
         self.__seek(offsets)
 
-    def commit(self) -> Mapping[TopicPartition, int]:
+    def stage_offsets(self, offsets: Mapping[TopicPartition, int]) -> None:
         if self.__state in {KafkaConsumerState.CLOSED, KafkaConsumerState.ERROR}:
             raise InvalidState(self.__state)
 
-        result: Optional[Sequence[ConfluentTopicPartition]] = None
+        if offsets.keys() - self.__working_offsets.keys():
+            raise ConsumerError("cannot stage offsets for unassigned streams")
 
-        retries_remaining = 3
-        while result is None:
-            try:
-                result = self.__consumer.commit(asynchronous=False)
-                assert result is not None
-            except KafkaException as e:
-                if not e.args[0].code() in (
-                    KafkaError.REQUEST_TIMED_OUT,
-                    KafkaError.NOT_COORDINATOR_FOR_GROUP,
-                    KafkaError._WAIT_COORD,
-                ):
-                    raise
-
-                if not retries_remaining:
-                    raise
-
-                logger.warning(
-                    "Commit failed: %s (%d retries remaining)",
-                    str(e),
-                    retries_remaining,
-                )
-                retries_remaining -= 1
-                time.sleep(1)
+        self.__staged_offsets.update(offsets)
 
-        offsets: MutableMapping[TopicPartition, int] = {}
+    def commit_offsets(self) -> Mapping[TopicPartition, int]:
+        if self.__state in {KafkaConsumerState.CLOSED, KafkaConsumerState.ERROR}:
+            raise InvalidState(self.__state)
 
-        for value in result:
-            # The Confluent Kafka Consumer will include logical offsets in the
-            # sequence of ``TopicPartition`` objects returned by ``commit``.
-            # These are an implementation detail of the Kafka Consumer, so we
-            # don't expose them here.
-            # NOTE: These should no longer be seen now that we are forcing
-            # offsets to be set as part of the assignment callback.
-            if value.offset in self.LOGICAL_OFFSETS:
-                continue
```

Comment:
These are no longer necessary, but it might make sense for us to validate offsets are non-negative wherever they are provided as input values just to be extra safe.